src/Inputs/SearchBar.tsx

```diff
@@ -0,0 +1,236 @@
+import React, {
+    useState,
+    useCallback,
+    useMemo,
+    Children,
+    useLayoutEffect,
+    useRef,
+    RefObject,
+} from 'react';
+import {
+    flex,
+    scroll,
+    position,
+    transition,
+    clickable,
+    darken,
+} from '@Utils/Mixins';
+
+import { LabelLayout, InputFragment as I, LabelLayoutProps } from '@Layouts';
+import { useTransition } from '@Utils/Hooks';
+
+import { Search } from 'styled-icons/fa-solid';
+import styled from 'styled-components';
+
+export interface SearchBarProps extends LabelLayoutProps {
+    suggestiveOptions?: string[];
+    value?: string | number;
+    onChange?: Function;
+    limit?: number;
+    numVisible: number;
+}
+
+export const SearchBar: React.FC<SearchBarProps> = ({
+    suggestiveOptions,
+    value,
+    limit = 4,
+    onChange = (): void => {},
+    name,
+    ...props
+}): React.ReactElement => {
+    const [expanded, setExpanded] = useState(false);
+    const [inputValue, setinputValue] = useState('');
+    const [, mount, animation] = useTransition(expanded);
+    const refSelectList = useRef() as RefObject<HTMLDivElement>;
```

Comment:
```suggestion
    const refSelectList = useRef<HTMLDivElement>();
```
Isn't this how the TypeScript works? I might be wrong, please check this.

Reply:
actually it doesn't work, Typescript gives error